freqtrade/strategy/interface.py

```diff
@@ -1618,18 +1646,49 @@ def ft_stoploss_reached(
 
         return ExitCheckTuple(exit_type=ExitType.NONE)
 
-    def min_roi_reached_entry(self, trade_dur: int) -> tuple[int | None, float | None]:
+    def min_roi_reached_entry(
+        self,
+        trade: Trade,
+        trade_dur: int,
+        current_time: datetime,
+    ) -> tuple[int | None, float | None]:
         """
         Based on trade duration defines the ROI entry that may have been reached.
         :param trade_dur: trade duration in minutes
         :return: minimal ROI entry value or None if none proper ROI entry was found.
         """
+
+        # Get custom ROI if use_custom_roi is set to True
+        if self.use_custom_roi:
+            custom_roi = strategy_safe_wrapper(
+                self.custom_roi, default_retval=None, supress_error=True
+            )(
+                pair=trade.pair,
+                trade=trade,
+                current_time=current_time,
```

Comment:
wouldn't we want trade_duration as parameter, too? 
so it's easy to get the minutes - making it simpler to relate to the traditional `minimal_roi` logic.